### ISSUE TITLE
Make "Show more" scroll back persistent?

### ISSUE THREAD

**Issue by winteriscariot** [diamondburned/dissent#328]
Is there a way to make sure any scrollback i reveal by clicking 'show more' sticks around perpetually for the session?

cuz i've noticed after a time the Dissent window will ... refresh, reset, or update or something, and it'll "chop off" the scroll back that i already revealed manually and go back to just the most recent however many lines of text

is there not a way to just write the scrollback to disk so it can be easily retrieved?

**Comment by diamondburned** [comment]
That's interesting. It shouldn't chop off unless you're scrolled to the bottom. Is it chopping off while you're still reading old messages?

**Comment by winteriscariot** [comment]
if i'm at the bottom of the chat history then yeah it'll chop off scrollback.

the issue i'm having is more that i have Dissent maximized on a vertical monitor (1920 px tall) with a lot of vertical space, and so when i go to a different chat only the top half of the chat has content:

![image](https://github.com/user-attachments/assets/5bcadf01-fe38-42d2-93f6-24e23bb1ebea)


is there a way to make that scrollback stick around? i'll notice that if i click "Show More", then leave it there (and i'm at the bottom of the scrollback) i'll notice the app sort of "refreshes" (i notice a flash in the corner of my eye) and it removed the expanded scrollback and goes back to the truncated chat log.

so basically, is it a big deal to make expanded scrollback not "refresh" and stick around?

**Comment by winteriscariot** [comment]
also it would be nice to make it so the expanded scrollback doesn't disappear when i switch between chats.

for instance i'll be in a DM, expand the scrollback, then flip to a server, then go back to the same DM, and the expanded scrollback will have vanished

it'd be cool if that scrollback that you expanded stayed around even after you switch between chats/servers

**Comment by diamondburned** [comment]
I think I make it clear out so quickly for RAM usage reasons. Once Go 1.24 comes out and I can fix the leak bugs in gotk4, I'll definitely improve this.
